bug: BDD NSFW フィルタがタグベース NSFW を認識しない

## 概要

`_apply_nsfw_filter` が Rating テーブルのみ検索するため、タグで NSFW 判定された画像が BDD シナリオで検出されない。

## 失敗テスト

- `tests/bdd/features/database_management.feature` の NSFW シナリオ (80-100行付近)

## エラー内容

\`\`\`
AssertionError: expected NSFW-tagged image to be in filter result
\`\`\`

## 根本原因

`src/lorairo/database/db_repository.py:2082-2113`:

\`\`\`python
def _apply_nsfw_filter(self, query, criteria):
    # Rating テーブル (rating="nsfw") のみ検索
    query = query.filter(Image.rating.has(Rating.rating == "nsfw"))
    # タグテーブルの "nsfw" タグは対象外
    return query
\`\`\`

タグ付けで "nsfw" を付けた画像は Rating テーブルには情報が入らないため、フィルタから漏れる。

## 修正方針

NSFW 判定を Rating + Tag の OR で実行:

\`\`\`python
def _apply_nsfw_filter(self, query, criteria):
    nsfw_by_rating = Image.rating.has(Rating.rating == "nsfw")
    nsfw_by_tag = Image.tags.any(Tag.name.in_(["nsfw", "explicit"]))
    query = query.filter(or_(nsfw_by_rating, nsfw_by_tag))
    return query
\`\`\`

タグ名の正規化 (大文字小文字、別名) はプロジェクトの既存ポリシーに従う。

## 関連ファイル

- `src/lorairo/database/db_repository.py:2082-2113`
- `tests/bdd/features/database_management.feature:80-100`

## 考慮事項

- NSFW タグのリストは設定で管理すべきか、定数で良いか
- 類似の safe/questionable/explicit 分類との整合性

--- a/docs/specs/civitai-content-levels.md
+++ b/docs/specs/civitai-content-levels.md
@@ -1,0 +1,22 @@
+# Civitai コンテンツ分類レベル
+
+出典: https://education.civitai.com/civitais-guide-to-content-levels/
+
+## 分類体系
+
+Civitai のすべてのコンテンツは映画レーティングに準拠した5段階で分類される。
+
+| レベル | 内容 |
+|--------|------|
+| PG     | Safe For Work。成人向けコンテンツなし |
+| PG-13  | 露出度の高い服装・セクシーな衣装、暴力、軽度のゴア |
+| R      | 成人向けテーマ・状況、部分的なヌード、グラフィックな暴力・死亡描写 |
+| X      | グラフィックなヌード、成人向けオブジェクト・設定 |
+| XXX    | 露骨な性的コンテンツ、または不穏なグラフィックコンテンツ |
+
+## プロジェクト内での使用箇所
+
+- `images.manual_rating` カラム（手動評価）
+- `ratings.normalized_rating` カラム（AI/手動編集モデルによるレーティング結果）
+- `_apply_nsfw_filter` での除外判定: `["r", "x", "xxx"]` を NSFW とみなす
+- UI の Rating 選択肢: `PG / PG-13 / R / X / XXX`

--- a/src/lorairo/database/db_repository.py
+++ b/src/lorairo/database/db_repository.py
@@ -2107,8 +2107,15 @@ class ImageRepository:
                 .correlate(Image)
             )
 
-            # AIレーティングまたは手動レーティングがNSFWである画像を除外
-            query = query.where(not_(or_(ai_nsfw_condition, manual_nsfw_condition)))
+            # タグベースのNSFW判定（"nsfw" / "explicit" タグが付いている画像を除外）
+            tag_nsfw_condition = (
+                exists()
+                .where(Tag.image_id == Image.id, func.lower(Tag.tag).in_(["nsfw", "explicit"]))
+                .correlate(Image)
+            )
+
+            # AIレーティング、手動レーティング、またはタグがNSFWである画像を除外
+            query = query.where(not_(or_(ai_nsfw_condition, manual_nsfw_condition, tag_nsfw_condition)))
             # レーティング情報がない (NULL) 画像は除外しない
         return query
 

